pkg/util/timeutil/timeutil.go

```diff
@@ -267,13 +268,19 @@ func FormatDurationStringDaysToHours(param string) (string, error) {
 // RoundToStartOfWeek creates a new time.Time for the preceding Monday 00:00 UTC
 func RoundToStartOfWeek(t time.Time) time.Time {
 	date := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, time.UTC)
+	if date.Weekday() == time.Sunday {
+		return date
+	}
 	daysFromMonday := int(date.Weekday())
 	return date.Add(-1 * time.Duration(daysFromMonday) * Day)
 }
 
 // RoundToStartOfFollowingWeek creates a new time.Time for the following Monday 00:00 UTC
 func RoundToStartOfFollowingWeek(t time.Time) time.Time {
 	date := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, time.UTC)
+	if date.Weekday() == time.Sunday {
+		return date
```

Comment:
Good question. The reason for this is to prevent a window=1w/step=1w query for a week into the future.

Some additional unit tests to ensure correct edge case functionality. The intended functionality is that '1w' means the current week, with additional weeks calculating backwards from there.

Reply:
Will echo Alex's concern here -- I would expect this function to return "the following Sunday" if I gave it a Sunday at noon. We should remember that this is a very general package, so others may use these functions for other reasons than the ones we have in mind. Will suggest we change this back, unless there's more context I'm missing.